modules/phase_field/src/materials/EpsAndMAndLAnisoGauss.C

```diff
@@ -0,0 +1,502 @@
+                                     (_qyR[a] * miubalibz_norm) - (_qzR[a] * miubaliby_norm);
+                const ADReal miuoy = (_qwR[a] * miubaliby_norm) - (_qxR[a] * miubalibz_norm) +
+                                     (_qyR[a] * 0.0) + (_qzR[a] * miubalibx_norm);
+                const ADReal miuoz = (_qwR[a] * miubalibz_norm) + (_qxR[a] * miubaliby_norm) -
+                                     (_qyR[a] * miubalibx_norm) + (_qzR[a] * 0.0);
+                ADReal miubalibwR = 0.0;
+                ADReal miubalibxR = miubalibx_norm;
+                ADReal miubalibyR = miubaliby_norm;
+                ADReal miubalibzR = miubalibz_norm;
+                if (_BoundaryNormal == 0)
+                {
+                  miubalibwR = miuow * _qwR[a] - miuox * _qxR[a] * -1.0 - miuoy * _qyR[a] * -1.0 -
+                               miuoz * _qzR[a] * -1.0;
+                  miubalibxR = miuow * _qxR[a] * -1.0 + miuox * _qwR[a] + miuoy * _qzR[a] * -1.0 -
+                               miuoz * _qyR[a] * -1.0;
+                  miubalibyR = miuow * _qyR[a] * -1.0 - miuox * _qzR[a] * -1.0 + miuoy * _qwR[a] +
+                               miuoz * _qxR[a] * -1.0;
+                  miubalibzR = miuow * _qzR[a] * -1.0 + miuox * _qyR[a] * -1.0 -
+                               miuoy * _qxR[a] * -1.0 + miuoz * _qwR[a];
+                }
+                _VlibR[_qp](0) = miubalibxR;
+                _VlibR[_qp](1) = miubalibyR;
+                _VlibR[_qp](2) = miubalibzR;
+                if (sumval == 0.0)
+                {
+                  _VlibR[_qp](0) = 0.0;
+                  _VlibR[_qp](1) = 0.0;
+                  _VlibR[_qp](2) = 0.0;
+                  miubalibwR = 0.0;
+                  miubalibxR = 0.0;
+                  miubalibyR = 0.0;
+                  miubalibzR = 0.0;
+                }
+
+                const ADReal dot_product =
+                    miubalibxR * vsmallbax + miubalibyR * vsmallbay + miubalibzR * vsmallbaz;
+                if (_thetabalib[l] < 0.0)
+                  _thetabalib[l] =
+                      6.28319 +
+                      _thetabalib[l]; // _thetabalib is in radians. 6.28319 radians is equivalent to
+                                      // 360 degrees. If the angle is negative, this converts it to
+                                      // its equivalent positive angle. -60deg around <111> =
+                                      // (360-60 = 300deg) around <111>.
+                const ADReal thetadtheta = thetaba - _thetabalib[l];
+                const ADReal normvbavalue = vbax * vbax + vbay * vbay + vbaz * vbaz;
+                const ADReal normvba = std::sqrt(normvbavalue);
+                const ADReal normvbalibvalue = _vbalibx[l] * _vbalibx[l] +
+                                               _vbaliby[l] * _vbaliby[l] +
+                                               _vbalibz[l] * _vbalibz[l];
+                const ADReal normvbalib = std::sqrt(normvbalibvalue);
```

Comment:
my comment on teh old code still applies here. Use vector operations and avoid componentwise wherever you can!

Reply:
See updated files